pkg/plugin/dumpstate.go

```diff
@@ -0,0 +1,255 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"github.com/neondatabase/autoscaling/pkg/api"
+	"github.com/neondatabase/autoscaling/pkg/util"
+)
+
+type dumpStateConfig struct {
+	Port           uint16 `json:"port"`
+	TimeoutSeconds uint   `json:"timeoutSeconds"`
+}
+
+func (c *dumpStateConfig) validate() (string, error) {
+	if c.Port == 0 {
+		return "port", errors.New("value must be > 0")
+	} else if c.TimeoutSeconds == 0 {
+		return "timeoutSeconds", errors.New("value must be > 0")
+	}
+
+	return "", nil
+}
+
+type stateDump struct {
+	Stopped   bool            `json:"stopped"`
+	BuildInfo util.BuildInfo  `json:"buildInfo"`
+	State     pluginStateDump `json:"state"`
+}
+
+func (p *AutoscaleEnforcer) startDumpStateServer(shutdownCtx context.Context) error {
```

Comment:
I think we don't actually use the shutdown context at all? is this an oversight?

Reply:
It's used in the call to `p.dumpState` below, `shutdownCtx.Err() != nil`